File: ruby/easy/shortest_word_distance.rb
# @param {String[]} words
# @param {String} word1
# @param {String} word2
# @return {Integer}
def shortest_distance(words, word1, word2)
    word1_index, word2_index = nil, nil
    min_distance = words.length
    words.each_with_index do |word, i|
        word1_index = i if word == word1
        word2_index = i if word == word2
        min_distance = [min_distance, (word1_index - word2_index).abs].min if word1_index && word2_index
    end
    return min_distance
end

File: javascript/easy/shortestWordDistance.js
/**
 * @param {string[]} words
 * @param {string} word1
 * @param {string} word2
 * @return {number}
 */
var shortestDistance = function(words, word1, word2) {
    let firstPointer = -1, secondPointer = -1, minDistance = words.length;
    for(let i = 0; i < words.length; i++) {
        if(words[i] === word1) {
            firstPointer = i;
        }
        if(words[i] === word2) {
            secondPointer = i;
        }
        if(firstPointer !== -1 && secondPointer !== -1) {
            minDistance = Math.min(minDistance, Math.abs(firstPointer - secondPointer));
        }
    }
    return minDistance;
};
